GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2017,6 +2012,23 @@ protected override void OnClosed(EventArgs e)
             base.OnClosed(e);
         }
 
+        private void CommandsToolStripMenuItem_DropDownOpening(object sender, System.EventArgs e)
+        {
+            //Most options do not make sense for artificial commits or no revision selected at all
+            var selectedRevisions = RevisionGrid.GetSelectedRevisions();
+            bool enabled = selectedRevisions.Count == 1 && !selectedRevisions[0].IsArtificial();
```

Comment:
Bug here! You forgot to check that is not bare repository

Reply:
Noted. I have planned another review after #4157 and at least one PR building on that is merged.
There are still some situations that should be disabled.

Are there any users with bare repositries? Why use a GUI at all?

Reply:
I rarely, just preferring use GUI for everything.